applets/wncklet/workspace-switcher.c

```diff
@@ -120,7 +120,8 @@ static void update_properties_for_wm(PagerData* pager)
 				gtk_widget_show(pager->workspace_names_scroll);
 			if (pager->display_workspaces_toggle)
 				gtk_widget_show(pager->display_workspaces_toggle);
-			g_object_set (pager->cell, "editable", TRUE, NULL);
+			if (pager->cell)
```

Comment:
```
cell = g_object_new(GTK_TYPE_CELL_RENDERER_TEXT, "editable", TRUE, NULL);
pager->cell = cell;
```

`g_object_new` does not return `NULL`. I think problem might be `applet_realized` function that seems like only way that could cause your problem...

So instead of adding extra NULL check I would say that code should be updated to make sure that this function is not called before everything is correctly setup. Same is true with other ifs in that switch.

Reply:
@muktupavels - good point, it seems `window_manager_changed` is called during `applet_realized`, except for the window manager hasn't changed - there's already listener for that. However, calling this avoids some duplicate code used in setting the window manager and initializing a few things, so those other `ifs` were most likely added for that same reason.

I think the code is not perfect like this, but my patch just follows the same methodology in place since this function was added (quite a while ago too: https://github.com/GNOME/gnome-panel/commit/250492f57753d9c3197ff27fee97038f957fb4e3).

So for now I'll leave it like this to fix the runtime error, but I'll add to my to-do list to refactor this file.